src/parseTools.js

```diff
@@ -118,6 +119,15 @@ function preprocess(text, filenameHint) {
             } else if (first === '#endif') {
               assert(showStack.length > 0);
               showStack.pop();
+            } else if (first === '#warning') {
+              if (showCurrentLine()) {
+                printErr(`${filenameHint}:${i+1}: #warning ${trimmed.substring(trimmed.indexOf(' ')).trim()}`);
+              }
+            } else if (first === '#error') {
+              if (showCurrentLine()) {
+                printErr(`${filenameHint}:${i+1}: #error ${trimmed.substring(trimmed.indexOf(' ')).trim()}`);
+                ++errorCount;
```

Comment:
We already have a global called `abortExecution` that you can set to true here.

It results in the compiler exiting with `Aborting compilation due to previous errors` at the end.

Reply:
great, updated to use that instead.